### PR TITLE
Add support for tuning leader election to the helm chart for  csi-provisioner and csi-attacher

### PR DESCRIPTION
**Pull request by moogzy**
**Is this a bug fix or adding new feature?**

Adding support to the helm chart for granular tuning of the leader election flags that the `csi-provisioner` and `csi-attacher` use.

Issue lodged: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/1368

**What is this PR about? / Why do we need it?**

The flags exist for tuning the behaviour of lease lock logic in the csi-provisioner and csi-attacher code. We have been observing lease issues due to a control plane issue that our manged provider is investigating. Therefore we want to tune these flags but need support for them in the helm chart.

This PR aims to implement this so that anyone using this chart can tune the flags when needed.

**What testing is done?** 

I've tested the chart logic with our local deployment:

// Default only `leader-election:true` enabeld (no diff).
```
 $ helmfile diff --context 1                                                                                                                                                                                                                                                                                                                                                                                                
Adding repo aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
"aws-ebs-csi-driver" has been added to your repositories

Building dependency release=aws-ebs-csi-driver, chart=../../../../../personal/oss/aws-ebs-csi-driver/charts/aws-ebs-csi-driver
Comparing release=aws-ebs-csi-driver, chart=../../../../../personal/oss/aws-ebs-csi-driver/charts/aws-ebs-csi-driver
```

// Adding flags successfully

```
$ helmfile diff --context 1                                                                                                                                                                                                                                                                                                                                                                                                
Adding repo aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
"aws-ebs-csi-driver" has been added to your repositories

Building dependency release=aws-ebs-csi-driver, chart=../../../../../personal/oss/aws-ebs-csi-driver/charts/aws-ebs-csi-driver
Comparing release=aws-ebs-csi-driver, chart=../../../../../personal/oss/aws-ebs-csi-driver/charts/aws-ebs-csi-driver
kube-system, ebs-csi-controller, Deployment (apps) has changed:
...
              - --leader-election=true
+             - --leader-election-lease-duration=20s
+             - --leader-election-renew-deadline=15s
+             - --leader-election-retry-period=10s
              - --default-fstype=ext4
...
```

// Disabling leader election, only configures the `leader-election=false` flag and doesn't add any others.

```
$ helmfile diff --context 1      
Adding repo aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
"aws-ebs-csi-driver" has been added to your repositories

Building dependency release=aws-ebs-csi-driver, chart=../../../../../personal/oss/aws-ebs-csi-driver/charts/aws-ebs-csi-driver
Comparing release=aws-ebs-csi-driver, chart=../../../../../personal/oss/aws-ebs-csi-driver/charts/aws-ebs-csi-driver
kube-system, ebs-csi-controller, Deployment (apps) has changed:
...
              - --v=2
-             - --leader-election=true
+             - --leader-election=false
            env:
...
```

### CODE REVIEW

**Review by torredil — CHANGES_REQUESTED** [review]
Thank you for your contribution and manually testing these changes.
/ok-to-test

File: charts/aws-ebs-csi-driver/values.yaml
Comment (moogzy) [reply charts/aws-ebs-csi-driver/values.yaml:198]: Hmm that's strange... I didn't make this change. Must have pulled it in from somewhere else. Will get rid of it.

File: charts/aws-ebs-csi-driver/values.yaml
Comment (moogzy) [reply charts/aws-ebs-csi-driver/values.yaml:47]: The reason I defaulted to true is because this was previously hardcoded to "true" in the helm template.

Thus if we default to false here then it breaks the default that most users of this template are relying on today.

File: charts/aws-ebs-csi-driver/values.yaml
Comment (moogzy) [reply charts/aws-ebs-csi-driver/values.yaml:28]: The reason I defaulted to true is because this was previously hardcoded to "true" in the helm template.

Thus if we default to false here then it breaks the default that most users of this template are relying on today.

File: charts/aws-ebs-csi-driver/values.yaml
Comment (moogzy) [reply charts/aws-ebs-csi-driver/values.yaml:25]: Done.

File: charts/aws-ebs-csi-driver/values.yaml
Comment (moogzy) [reply charts/aws-ebs-csi-driver/values.yaml:29]: Done.

File: charts/aws-ebs-csi-driver/values.yaml
Comment (torredil) [reply charts/aws-ebs-csi-driver/values.yaml:47]: I see, you're totally right - let's leave it as `true` in that case.

File: charts/aws-ebs-csi-driver/values.yaml
Comment (torredil) [inline charts/aws-ebs-csi-driver/values.yaml:48]: final nit: behavior. Also, please squash the commits before we merge this in. Other than that, lgtm. I will approve this PR once another maintainer has reviewed these changes.

Comment (moogzy) [reply charts/aws-ebs-csi-driver/values.yaml:48]: Oh dear..... I thought I got them all. Thank you!

Comment (moogzy) [reply charts/aws-ebs-csi-driver/values.yaml]: Commits squashed. Thanks for the review!

**Comment by moogzy** [comment]
@rdpsin @gtxu 

Hello, would it be possible to complete a review of this PR please?

**Comment by moogzy** [comment]
@torredil I'm not getting a response from the other reviewers on here or on slack. Would you be comfortable approving or do we need an additional reviewer?

**Review by ConnorJC3 — APPROVED** [review]
lgtm

**Review by torredil — APPROVED** [review]

**Pull request merged**
